Homework for sprint №23 (App icon, playlist info&edit screen)

sobolevkir opened the pull request.

sobolevkir committed:
Добавлена иконка приложения.
Также скорректированы стили.

sobolevkir committed:
Реализован экран просмотра плейлиста. На данном экране показана обложка, длительность плейлиста, количество треков в нём, а также список треков.
Любой из треков можно удалить долгим нажатием на него.

sobolevkir committed:
Доработан экран просмотра плейлиста:
добавлен функционал редактирования плейлиста, его удаления, а также теперь плейлистом можно поделиться во внешнем приложении.

sobolevkir committed:
Небольшие исправления:
ненужные файлы обложек теперь удаляются после удаления плейлиста или изменения обложки в существующем плейлисте;
скорректированы цвета текста на экране информации о плейлисте в тёмной теме;
кнопка добавления в плейлист на экране плеера теперь изменяет свою иконку в случае, если трек уже добавлен в какой-либо существующий плейлист.

Charmandik commented on .idea/kotlinc.xml:
Попробуй добавить этот тип файлов в гитигнор, их не обязательно отслеживать в гите, они генерируемые и у каждого свои

Charmandik commented on app/src/main/java/com/sobolevkir/playlistmaker/playlists/ui/PlaylistCreateFragment.kt:
```diff
@@ -17,27 +17,27 @@ import com.bumptech.glide.load.resource.bitmap.CenterCrop
-    private val viewModel: CreatePlaylistViewModel by viewModel()
+    protected val binding by viewBinding(FragmentPlaylistCreateBinding::bind)
+    protected open val viewModel: PlaylistCreateViewModel by viewModel()
     private lateinit var pickMedia: ActivityResultLauncher<PickVisualMediaRequest>
```
Вместо использования lateinit, я рекомендую тебе использовать подход с добавлением nullability к типу объекта с оператором ?. и сразу проинициализировать его значением null. Это решение позволяет избежать ошибок, связанных с попытками доступа к неинициализированным объектам, которые могут привести к непредсказуемому поведению или крашам приложения. Хотя это может добавить некоторую дополнительную работу при обращении к объекту через оператор ?. , но это значительно повысит безопасность приложения, так как ты будешь уверен, что объект всегда либо имеет значение, либо равен null, и не будет вызывать исключительных ситуаций.Можешь еще прочитать статью на эту тему https://habr.com/ru/companies/maxilect/articles/590061

Charmandik commented on app/src/main/java/com/sobolevkir/playlistmaker/favorites/data/impl/FavoritesRepositoryImpl.kt:
```diff
@@ -16,8 +16,8 @@ class FavoritesRepositoryImpl(
+        appDatabase.getFavoriteTrackDao().removeTrack(trackId)
     }
 
     override fun getFavoriteTracks(): Flow<List<Track>> =
```
В Kotlin лучше использовать "return" вместо "=", потому что это помогает делать код более читаемым и явным. Когда ты используешь "return", это явно указывает, что функция возвращает определенное значение. В то же время, использование "=" может вызывать путаницу, особенно если в функции есть несколько выражений.

Charmandik commented on app/src/main/res/layout/layout_playlist_small_list_item.xml:
```diff
@@ -42,7 +42,7 @@
             android:textAppearance="@style/YS_Regular_11"
             android:textColor="?attr/colorPrimaryVariant"
-            tools:text="@string/hint_tracks_number" />
+            tools:text="@string/hint_playlist_tracks_number" />
```
чтобы самому было проще понимать, где в верстке текст, который отображается только в превью студии, а где текст, который "зашит", рекомендую текста, используемые в атрибуте tools, оборачивать в квадратные скобки.

Charmandik commented on app/src/main/java/com/sobolevkir/playlistmaker/playlists/ui/PlaylistInfoFragment.kt:
```diff
@@ -0,0 +1,242 @@
+    }
+
+    companion object {
+        private const val CLICK_DEBOUNCE_DELAY = 100L
```
При именовании констант для промежутка времени рекомендую всегда в конце добавлять единицу времени, например _MILLIS, _IN_SECONDS, _HOURS и т.д.

sobolevkir committed:
Небольшие исправления:
скорректирована анимация перехода между теми фрагментами, в которых есть BottomNavigationView и теми, в которых его нет, теперь нет эффекта "прыжка";
скорректированы имена констант.
